api/app/signals/apps/signals/utils/map.py

```diff
@@ -0,0 +1,78 @@
+import io
+import urllib.request
+from math import asinh, pi, radians, tan
+
+from PIL import Image
+
+TILE_SIZE = 256
+
+
+#  https://wiki.openstreetmap.org/wiki/Slippy_map_tilenames#Tile_numbers_to_lon..2Flat._3
+class MapGenerator:
+    def _deg2num(self, lat_deg, lon_deg, zoom):
+        lat_rad = radians(lat_deg)
+        n = 2.0 ** zoom
+        xtile = (lon_deg + 180.0) / 360.0 * n
+        ytile = (1.0 - asinh(tan(lat_rad)) / pi) / 2.0 * n
+        return (xtile, ytile)
+
+    def deg2num(self, lat_deg, lon_deg, zoom):
+        (xtile, ytile) = self._deg2num(lat_deg, lon_deg, zoom)
```

Comment:
Style nitpick; leave out the parentheses in the left hand side. (Several instances more in this PR.)

Reply:
check